### ISSUE TITLE
[Bug]: Warning is shown: The CJS build of Vite's Node API is deprecated

### ISSUE THREAD

**Issue by IanVS** [storybookjs/storybook#24333]
### Describe the bug

With the latest Vite 5 beta (beta.3), we are getting a warning logged to the terminal when starting up storybook:

```
The CJS build of Vite's Node API is deprecated. See https://vitejs.dev/guide/troubleshooting.html#vite-cjs-node-api-deprecated for more details.
```

I guess they're going to force us to start using their ESM build, and we should probably do it sooner than later, to avoid users reporting this warning.

### To Reproduce

https://stackblitz.com/edit/github-wgldvd?file=package.json

### System

```bash
N/A
```

**Comment by bentaly** [comment]
Hi, out of curiosity, will this change be included in version 8? given it seems support for Vite 5 will be dealt with in that release?
https://github.com/storybookjs/storybook/issues/25115
If so is there a rough timescale?

Thanks

**Comment by IanVS** [comment]
Vite 5 is fully supported in Storybook 7, though in some cases we are using the Vite CJS APIs.  They continue to work just fine, and we will be sure to remove usage of them in Storybook 8 so that when they are removed in Vite 6 or above, Storybook continues to work without a problem.

**Comment by jeremytenjo** [comment]
If anybody still has this issue, I fix it by updating Vite to `"vite": "7.2.4",`
